packages/runtime/src/runtime.ts

```diff
@@ -62,55 +63,67 @@ export class Runtime {
    */
-  assertAppDefined (appId: number, line?: number): SSCAttributesM {
+  assertAppDefined (app: SSCAttributesM | undefined, appId: number, line?: number): SSCAttributesM {
     const lineNumber = line ?? 'unknown';
-    const app = this.ctx.state.globalApps.get(appId);
     if (app === undefined) {
       throw new TealError(ERRORS.TEAL.APP_NOT_FOUND, { appId: appId, line: lineNumber });
     }
     return app;
   }
 
   /**
-   * Fetch account using accountIndex from `Accounts` list
-   * Accounts: List of accounts in addition to the sender
-   * that may be accessed from the application's approval-program and clear-state-program.
-   * @param accountIndex index of account to fetch from account list
-   * @param line line number
-   * NOTE: index 0 represents txn sender account
-   * - if user is accessing this function directly through runtime, the line number is unknown
+   * Fetches app from `this.store`
+   * @param appId Application Index
    */
-  getAccount (accountIndex: bigint, line?: number): StoreAccountI {
-    let account: StoreAccountI | undefined;
-    if (accountIndex === BIGINT0) {
-      const senderAccount = encodeAddress(this.ctx.tx.snd);
-      account = this.ctx.state.accounts.get(senderAccount);
-    } else {
-      const accIndex = accountIndex - BIGINT1;
-      checkIndexBound(Number(accIndex), this.ctx.tx.apat, line);
-      const pkBuffer = this.ctx.tx.apat[Number(accIndex)];
-      account = this.ctx.state.accounts.get(encodeAddress(pkBuffer));
+  getApp (appId: number): SSCAttributesM {
+    if (!this.store.globalApps.has(appId)) {
+      throw new TealError(ERRORS.TEAL.APP_NOT_FOUND, { appId: appId, line: 'unknown' });
     }
+    const accAddress = this.assertAddressDefined(this.store.globalApps.get(appId));
+    const account = this.store.accounts.get(accAddress);
+    const app = account?.createdApps.get(appId);
+    return this.assertAppDefined(app, appId);
+  }
+
+  /**
+   * Fetches account from `this.store`
+   * @param address account address
+   */
+  getAccount (address: string): StoreAccountI {
+    const account = this.store.accounts.get(address);
     return this.assertAccountDefined(account);
   }
 
   /**
-   * Fetches global state value for key present app's global data
-   * returns undefined otherwise
+   * Fetches global state value for key present in creator's global state
+   * for given appId, returns undefined otherwise
    * @param appId: current application id
    * @param key: key to fetch value of from local state
    */
   getGlobalState (appId: number, key: Uint8Array | string): StackElem | undefined {
-    // TODO: will be updated in https://www.pivotaltracker.com/story/show/176487715
-    // we will operate on accounts rather than globalApp map
-    const app = this.assertAppDefined(appId);
+    const app = this.getApp(appId);
     const appGlobalState = app["global-state"];
     const globalKey = keyToBytes(key);
     return appGlobalState.get(globalKey.toString());
```

Comment:
Same as above. `runtime.getGlobalState => calls account.getGlobalState`. We can check the getLocalState function below (I think we discussed these updates in the `improve DevUx` task, but at that time we were operating on `globalApps` only and not `store.accounts`.